### ISSUE TITLE
Nodester app init issue

### ISSUE THREAD

**Issue by chrismatthieu** [nodester/nodester#96]
Nodester app init returns nodester ERRO false running node.js 0.4.3 and current nodester-cli

christoffe:node-v0.4.3 christoffe$ node -v
v0.4.3

christoffe:node-v0.4.3 christoffe$ cd ..

christoffe:src christoffe$ npm install nodester-cli
npm info it worked if it ends with ok
npm info using npm@0.3.15
npm info using node@v0.4.3
npm info preinstall nodester-cli@0.2.12
npm info install nodester-cli@0.2.12
npm info postinstall nodester-cli@0.2.12
npm info predeactivate nodester-cli@0.2.12
npm info deactivate nodester-cli@0.2.12
npm info postdeactivate nodester-cli@0.2.12
npm info preactivate nodester-cli@0.2.12
npm info activate nodester-cli@0.2.12
npm info postactivate nodester-cli@0.2.12
npm info build Success: nodester-cli@0.2.12
npm ok

christoffe:src christoffe$ nodester app create clm405 hello.js
nodester info creating app: clm405 hello.js
nodester info successfully created app clm405 to will run on port 9033 from hello.js
nodester info run nodester app init clm405 to setup this app.
christoffe:src christoffe$ nodester app init clm405
nodester info initializing git repo for clm405 into folder clm405
nodester info cloning the repo git clone ec2-user@nodester.com:/node/hosted_apps/topher/880-a483a45ddb3ef9aa28ee9a155b02e38f.git clm405
Warning: the RSA host key for 'nodester.com' differs from the key for the IP address '50.16.203.53'
Offending key for IP in /Users/christoffe/.ssh/known_hosts:1
Matching host key in /Users/christoffe/.ssh/known_hosts:14
Are you sure you want to continue connecting (yes/no)? yes
nodester info clone complete
nodester info writing app data to config
nodester info writing app files
nodester info processing the initial commit
Warning: the RSA host key for 'nodester.com' differs from the key for the IP address '50.16.203.53'
Offending key for IP in /Users/christoffe/.ssh/known_hosts:1
Matching host key in /Users/christoffe/.ssh/known_hosts:14
Are you sure you want to continue connecting (yes/no)? yes
nodester info attemping to start the new app.
nodester ERRO false

**Comment by kenkam** [comment]
Hi Chris,

I am having the same issue. I was about to start a new ticket but conscience told me to look at existing ones first. Here's what I did:
1. nodester user setup user pass
2. nodester app create kenkamapp
3. nodester app init kenkamapp

To which I get an error message, same as the above:

```
nodester info attemping to start the new app.
nodester ERRO false
```

Alright, so I try a `nodester app start kenkamapp`, which gives the same error.

I then cd into ./kenkamapp to see what's actually in there. The hello.js file is there and a git remote show reveals that origin is already setup as `ec2-user@nodester.com:/node/hosted_apps/kenkam/893-1...1e5.git`.

I touch a test file and I add and commit locally. I then try to git push origin master, which gave:

```
error: insufficient permission for adding an object to repository database ./objects
```

and the push fails.

**Comment by richardanaya** [comment]
This is the exact same issue i'm having too

**Comment by uchiyama2000** [comment]
maybe i'm having too.

<pre>
% node -v
v0.4.3

% nodester app init a
nodester info initializing git repo for a into folder a
nodester info cloning the repo git clone ec2-user@nodester.com:/node/hosted_apps/username/xxxxxxxxxxxxxxxx.git a
nodester info clone complete
nodester info writing app data to config
fs.js:221
  return binding.open(path, stringToFlags(flags), mode);
                 ^
Error: EBADF, Bad file descriptor 'a/.nodester.appconfig'
    at Object.openSync (fs.js:221:18)
    at Object.writeFileSync (fs.js:473:15)
    at Object.writeApp (/Users/xxxxxxx/.node_libraries/.npm/nodester-cli/0.2.12/package/lib/config.js:54:8)
    at /Users/uchiyama/.node_libraries/.npm/nodester-cli/0.2.12/package/lib/app.js:230:37
    at ChildProcess.exithandler (child_process.js:106:7)
    at ChildProcess.emit (events.js:67:17)
    at ChildProcess.onexit (child_process.js:192:12)
</pre>

**Comment by kevinthompson** [comment]
Also having this problem:

```
kevinthompson:helper ~/projects $ nodester app create bitchat
nodester info creating app: bitchat server.js
nodester info successfully created app bitchat to will run on port 9079 from server.js
nodester info run nodester app init bitchat to setup this app.
kevinthompson:helper ~/projects $ nodester app init bitchat
nodester info initializing git repo for bitchat into folder bitchat
nodester info cloning the repo git clone ec2-user@nodester.com:/node/hosted_apps/kevinthompson/926-75a5a93485c35493d57198fca64e9d34.git bitchat
nodester info clone complete
nodester info writing app data to config
nodester info writing app files
nodester info processing the initial commit
nodester info attemping to start the new app.
nodester ERRO false
```

**Comment by chrismatthieu** [comment]
Hi Kevin,

A few of us are having the same issue at them moment.  I have opened the
following ticket to track it.

https://github.com/nodester/nodester/issues/96

Regards,
Chris

**Comment by chrismatthieu** [comment]
After a little more research, it looks like the app isn't actually being created properly before the app init.  If you run nodester app info appname after creating the app, the CLI returns app not found.

christoffe:nodetest321 christoffe$ nodester app info hello2311
nodester info Gathering information about: hello2311
nodester ERRO [HTTP 400] {"status" : "failure - app not found (" + appname + ")"}

**Comment by kenkam** [comment]
Chris,

I'm getting something slightly different.

```
$ nodester app create foobar
$ nodester app info foobar
nodester info Gathering information about: foobar
nodester warn foobar on 1234 running: false (pid: unknown)
...

$ nodester app init foobar
// error as before
...
nodester ERRO false

$ nodester app info foobar
nodester info Gathering information about: foobar
nodester warn foobar on 1234 running: failed-to-start
...
```

**Comment by chrismatthieu** [comment]
After upgrading express to the latest version 2.1.0 and reinstalling requires for couch db, all of the REST APIs started working again!

**Comment by kenkam** [comment]
Nice work. I tried it straight away but now when I try to do `app init` with a newly created app, I get a EBADF error:

```
Error: EBADF, Bad file descriptor 'atestapp123/.nodester.appconfig'
at Object.openSync (fs.js:221:18)
at Object.writeFileSync (fs.js:473:15)
at Object.writeApp (/usr/local/lib/node/.npm/nodester-cli/0.2.12/package/lib/config.js:54:8)
at /usr/local/lib/node/.npm/nodester-cli/0.2.12/package/lib/app.js:230:37
at ChildProcess.exithandler (child_process.js:106:7)
at ChildProcess.emit (events.js:67:17)
at ChildProcess.onexit (child_process.js:192:12)
```

**Comment by chrismatthieu** [comment]
I am running node.js 0.4.3.  Could you verify that you are also running 0.4.3 or later?

**Comment by richardanaya** [comment]
Thank you! It worked for me.  I did the following installs and deleted and recreated my app.

npm install express
npm install promised-io
npm install traits
npm install couchdb

nodester delete app my_app_name
nodester create app my_app_name

**Comment by kenkam** [comment]
Yes, I am on v0.4.3 on Mac OS X. I built node on this machine and I can verify it works correctly.

**Comment by abeyonalaja** [comment]
Hi Chris, I am on v0.4.4 and I am still getting the same error

nodester info initializing git repo for abeynodesterapp into folder abeynodesterapp
nodester info cloning the repo git clone ec2-user@nodester.com:/node/hosted_apps/abey/980-6e3d4d6b4b48cef072f154805aaec144.git abeynodesterapp
nodester info clone complete
nodester info writing app data to config

fs.js:221
  return binding.open(path, stringToFlags(flags), mode);
                 ^
Error: EBADF, Bad file descriptor 'abeynodesterapp/.nodester.appconfig'
    at Object.openSync (fs.js:221:18)
    at Object.writeFileSync (fs.js:473:15)
    at Object.writeApp (/usr/local/lib/node/.npm/nodester-cli/0.2.12/package/lib/config.js:54:8)
    at /usr/local/lib/node/.npm/nodester-cli/0.2.12/package/lib/app.js:230:37
    at ChildProcess.exithandler (child_process.js:106:7)
    at ChildProcess.emit (events.js:67:17)
    at ChildProcess.onexit (child_process.js:192:12)

**Comment by muellersfranz** [comment]
Hi Chris, 
having the same problem..

$  nodester create app dog

nodester info creating app: dog server.js
nodester info successfully created app dog to will run on port 9157 from server.js
nodester info run nodester app init dog to setup this app.

$ nodester app init dog 

...

nodester info writing app data to config

fs.js:221
  return binding.open(path, stringToFlags(flags), mode);
                 ^
Error: EBADF, Bad file descriptor 'dog/.nodester.appconfig'
    at Object.openSync (fs.js:221:18)
    at Object.writeFileSync (fs.js:512:15)
    at Object.writeApp (/usr/local/lib/node/.npm/nodester-cli/0.2.12/package/lib/config.js:54:8)
    at /usr/local/lib/node/.npm/nodester-cli/0.2.12/package/lib/app.js:230:37
    at ChildProcess.exithandler (child_process.js:106:7)
    at ChildProcess.emit (events.js:67:17)
    at ChildProcess.onexit (child_process.js:192:12)

thx!

**Comment by kenkam** [comment]
I think this issue has now been fixed! I did a `app create` and `app init` and it works fine. Great work guys!

**Comment by chrismatthieu** [comment]
Fixed by @DavGlass

**Comment by ghost** [comment]
I'm getting the same exact "Bad file descriptor" errors as others, and I'm using node.js 0.4.7 and nodester-cli 0.2.12 – any advice?

**Comment by chrismatthieu** [comment]
Do you have a passphrase on your rsa key? Are you using the same computer that you used to create your account?

**Comment by ghost** [comment]
Yes, and yes. I don't have a lot of experience with public keys, so is there something I'm doing wrong?

**Comment by chrismatthieu** [comment]
You need to generate a new RSA key on your machine without a passphrase and add it to Nodester using:

nodester user setkey

**Comment by ghost** [comment]
I followed your instructions and I'm still getting the exact same error – any ideas?

**Comment by jswartwood** [comment]
Can you not use the noderster app create on existing code?  I had the same problem when attempting such a thing... I tried again after the app create step to go back to curl to setup my git remote.  (Also, git push complained about my ssh key, so I ran the curl rsakey= line.)  http://nodester.com/api.html#userapi Wonder if the curl commands work for anyone else having this problem?

**Comment by chrismatthieu** [comment]
The steps for deploying an existing app are on the nodester home page. You basically do the following:

Nodester app create myappname
Nodester app info myappname
Copy git repo string 
Git remote add nodeter pasterepostring
Git push nodester master

Don't forget to nodester NPM install modules and restart if needed.

**Comment by jswartwood** [comment]
Ah... I obviously skimmed a little too quickly :)  I figured it out with a bit of tinkering though.  Thanks for posting that clarification; (my error looked exactly the same as some of the others posted) hopefully it will help some people.  I'm excited to compare nodester with my experiences using duostack and no.de.

**Comment by barbagallo** [comment]
When I try to git push nodester master, I get:

Connection closed by 50.16.203.53
fatal: The remote end hung up unexpectedly

Strange, and my keys match up on my machine and the nodester backend..
